### PR TITLE
UF-8359: Created integrations to OeP

### PR DESCRIPTION
**Pull request by MickeWallner**

### CODE REVIEW

File: src/main/java/se/sundsvall/smloader/integration/openeexternal/OpenEExternalClient.java [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+import io.github.resilience4j.circuitbreaker.annotation.CircuitBreaker;
+import org.springframework.cloud.openfeign.FeignClient;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestParam;
+import se.sundsvall.smloader.integration.openeexternal.configuration.OpenEExternalConfiguration;
+
+import static se.sundsvall.smloader.integration.openeexternal.configuration.OpenEExternalConfiguration.CLIENT_ID;
+
+@FeignClient(name = CLIENT_ID, url = "${integration.open-e-internal.url}", configuration = OpenEExternalConfiguration.class)
```
Comment (ronnberg) [inline src/main/java/se/sundsvall/smloader/integration/openeexternal/OpenEExternalClient.java:12]: integration.open-e-internal.url on OpenEExternalClient?

File: src/main/java/se/sundsvall/smloader/integration/openeexternal/configuration/OpenEExternalConfiguration.java [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+public class OpenEExternalConfiguration {
+
+	public static final String CLIENT_ID = "open-e-external";
+
+	@Bean
+	FeignBuilderCustomizer feignBuilderCustomizer(final OpenEExternalProperties properties, ClientRegistrationRepository clientRegistrationRepository) {
+		return FeignMultiCustomizer.create()
+			.withErrorDecoder(new ProblemErrorDecoder(CLIENT_ID))
+			.withRequestTimeoutsInSeconds(properties.connectTimeout(), properties.readTimeout())
+			.withRetryableOAuth2InterceptorForClientRegistration(clientRegistrationRepository.findByRegistrationId(CLIENT_ID))
```
Comment (ronnberg) [inline src/main/java/se/sundsvall/smloader/integration/openeexternal/configuration/OpenEExternalConfiguration.java:21]: Oauth2 retry needed?

File: src/main/java/se/sundsvall/smloader/integration/openeinternal/configuration/OpenEInternalConfiguration.java [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+public class OpenEInternalConfiguration {
+
+	public static final String CLIENT_ID = "open-e-internal";
+
+	@Bean
+	FeignBuilderCustomizer feignBuilderCustomizer(final OpenEInternalProperties properties, ClientRegistrationRepository clientRegistrationRepository) {
+		return FeignMultiCustomizer.create()
+			.withErrorDecoder(new ProblemErrorDecoder(CLIENT_ID))
+			.withRequestTimeoutsInSeconds(properties.connectTimeout(), properties.readTimeout())
+			.withRetryableOAuth2InterceptorForClientRegistration(clientRegistrationRepository.findByRegistrationId(CLIENT_ID))
```
Comment (ronnberg) [inline src/main/java/se/sundsvall/smloader/integration/openeinternal/configuration/OpenEInternalConfiguration.java:21]: Oauth2 retry needed?

File: src/test/java/se/sundsvall/smloader/integration/openeexternalsoap/configuration/OpenEExternalSoapConfigurationTest.java [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+
+		try (final MockedStatic<FeignMultiCustomizer> feignMultiCustomizerMock = Mockito.mockStatic(FeignMultiCustomizer.class)) {
+			feignMultiCustomizerMock.when(FeignMultiCustomizer::create).thenReturn(feignMultiCustomizerSpy);
+
+			final var customizer = configuration.feignBuilderCustomizer(propertiesMock);
+
+			final ArgumentCaptor<OpenESoapErrorDecoder> errorDecoderCaptor = ArgumentCaptor.forClass(OpenESoapErrorDecoder.class);
+			final ArgumentCaptor<SOAPEncoder> soapEncoderCaptor = ArgumentCaptor.forClass(SOAPEncoder.class);
+			final ArgumentCaptor<SOAPDecoder> soapDecoderCaptor = ArgumentCaptor.forClass(SOAPDecoder.class);
+
```
Comment (ronnberg) [inline src/test/java/se/sundsvall/smloader/integration/openeexternalsoap/configuration/OpenEExternalSoapConfigurationTest.java:59]: verify withRequestInterceptor?

**Review by cheezi0747 — APPROVED** [review]

**Pull request merged**